R/GenerateCovariateReport.R

```diff
@@ -18,12 +20,270 @@
+  
+  GenerateDataReport <- function(input_data, output_file = "data_report.html", output_dir = getwd(), ...) {
+    ## Get argument list
+    args <- as.list(match.call())
+    ## Get directory of report markdown template
+    data_report_dir <- tempfile(fileext = ".rmd", tmpdir = tempdir())
+    cat(DataReportText, file = data_report_dir, fill = 1)
+    ## Render report into html
+    render(
+      input = data_report_dir,
+      output_file = output_file,
+      output_dir = output_dir,
+      intermediates_dir = output_dir,
+      params = list(data = input_data, fun_options = list()),
+      ...
+    )
+    ## Open report
+    report_path <- file.path(output_dir, output_file)
+    browseURL(report_path)
+    ## Print report directory
+    if (ifelse(is.null(args[["quiet"]]), TRUE, !args[["quiet"]])) message(paste0("\n\nReport is generated at \"", report_path, "\"."))
+  }
+  
+  GenerateRasterReport <- function(input_data, output_file = "raster_report.html", output_dir = getwd(), ...) {
+    ## Get argument list
+    args <- as.list(match.call())
+    ## Get directory of report markdown template
+    raster_report_dir <- tempfile(fileext = ".rmd", tmpdir = tempdir())
+    cat(RasterReportText, file = raster_report_dir, fill = 1)
+    ## Render report into html
+    render(
+      input = raster_report_dir,
+      output_file = output_file,
+      output_dir = output_dir,
+      intermediates_dir = output_dir,
+      params = list(data = input_data, fun_options = list()),
+      ...
+    )
+    ## Open report
+    report_path <- file.path(output_dir, output_file)
+    browseURL(report_path)
+    ## Print report directory
+    if (ifelse(is.null(args[["quiet"]]), TRUE, !args[["quiet"]])) message(paste0("\n\nReport is generated at \"", report_path, "\"."))
+  }
+  
+  # if(type != any("D","R","DR")){
+  #   stop ('Type must be one of "D", "R", or "DR"')
+  # }
+  
+  ###################################################
+  ### Generate reports dependent on selected type ###
+  ###################################################
   
-  GenerateReport(data.table(as.data.frame(.ras)))
+  cov_names <- attr(.model$data, 'covCols')   # extract attribute of .model$data saying which colums are covariates
   
+  if(type == "D"){   # generate data report only
+    print(paste0("Generate data report"))
+    GenerateDataReport(data.table(as.data.frame(.model$data[,cov_names, drop = FALSE])))
+  } else {
+    if(type == "R"){   # generate raster report only
+      print(paste0("Generate raster report"))
+      GenerateRasterReport(data.table(as.data.frame(.ras)))
+    } else {
+      if(type == "DR"){   # generate data and raster report
+        print(paste0("Generate data and raster report"))
+        GenerateDataReport(data.table(as.data.frame(.model$data[,cov_names, drop = FALSE])))
+        GenerateRasterReport(data.table(as.data.frame(.ras)))
+      }
+    }
+  }
+  unlink(Sys.glob("file*"), force = TRUE, recursive = TRUE)
```

Comment:
Fair call. `tempdir()` won't work because the `output_dir` is where the reports get saved. I still can't figure out what is generating these empty folders that have nothing to go with what's going on but I imagine something like this would get around that:

dir.create("Reports")
switch both output_dir commands to "Reports" instead of getwd()
change the unlink command to "Reports/file*"